### PR TITLE
Fix numpy warnings

### PR DESCRIPTION
**Pull request by ssanderson**
Toward fixing https://github.com/quantopian/zipline/issues/954.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -33,43 +34,65 @@
...
+            """
+            def build_extensions(self):
+                """
+                Lazily append numpy's include directory to Extension includes.
+
+                This is done here rather than at module scope because setup.py
+                may be run before numpy has been installed, in which case
+                importing numpy and calling `numpy.get_include()` will fail.
+                """
+                numpy_incl = resource_filename('numpy', 'core/include')
```
Comment (llllllllll) [inline setup.py]: how is this different from `numpy.get_include()`

File: zipline/utils/numpy_utils.py [hunk 2]
```diff
@@ -22,11 +22,19 @@
 
 make_datetime64ns = flip(datetime64, 'ns')
 make_datetime64D = flip(datetime64, 'D')
-np_NaT = make_datetime64ns('NaT')
+
+NaTmap = {
+    dtype('datetime64[%s]' % unit): datetime64('NaT', unit)
+    for unit in ('ns', 'us', 'ms', 's', 'm', 'D')
+}
+NaT_for_dtype = NaTmap.__getitem__
```
Comment (llllllllll) [inline zipline/utils/numpy_utils.py]: `toolz.curried.operator.getitem(NaTmap)` maybe?

File: setup.py [hunk 3]
```diff
@@ -33,43 +34,65 @@
...
+                for ext in self.extensions:
+                    ext.include_dirs.append(numpy_incl)
+
+                # This explicitly calls the superclass method rather than the
+                # usual super() invocation because distutils' build_class, of
+                # which Cython's build_ext is a subclass, is an old-style class
+                # in Python 2, which doesn't support `super`.
+                cython_build_ext.build_extensions(self)
```
Comment (llllllllll) [inline setup.py]: get around this by subclassing `cython_build_ext` and `object`.

File: zipline/data/loader.py [hunk 4]
```diff
@@ -239,8 +240,15 @@ def ensure_benchmark_data(symbol, first_date, last_date, now, trading_day):
...
-    data.to_csv(path)
+    try:
+        data = get_benchmark_returns(
+            symbol,
+            first_date - trading_day,
+            last_date,
+        )
+        data.to_csv(path)
+    except (OSError, IOError, HTTPError):
+        logger.exception('falied to cache the new benchmark returns')
```
Comment (richafrank) [inline zipline/data/loader.py]: typo "failed"

File: zipline/utils/numpy_utils.py [hunk 5]
```diff
@@ -22,11 +22,19 @@
...
+NaTns = NaT_for_dtype(datetime64ns_dtype)
+NaTD = NaT_for_dtype(datetime64D_dtype)
+
 
 _FILLVALUE_DEFAULTS = {
     float64_dtype: nan,
-    datetime64ns_dtype: np_NaT,
+    datetime64ns_dtype: NaT_for_dtype(datetime64ns_dtype),
```
Comment (richafrank) [inline zipline/utils/numpy_utils.py]: Can reuse above const

File: zipline/utils/numpy_utils.py [hunk 6]
```diff
@@ -142,8 +153,8 @@ def busday_count_mask_NaT(begindates, enddates, out=None):
     if out is None:
         out = empty(broadcast(begindates, enddates).shape, dtype=float)
 
-    beginmask = (begindates == np_NaT)
-    endmask = (enddates == np_NaT)
+    beginmask = (begindates == NaT)
+    endmask = (enddates == NaT)
```
Comment (richafrank) [inline zipline/utils/numpy_utils.py]: @ssanderson Should we use the dtype from begin/enddates?

Comment (ssanderson) [inline zipline/utils/numpy_utils.py]: `busday_count` only supports `datetime64[D]`, so this should only ever be `NaTD`:

```
In [6]: busday_count(arange(0, 3, dtype='datetime64[ns]'), arange(0, 3, dtype='datetime64[ns]'))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-3e0be598dbd6> in <module>()
----> 1 busday_count(arange(0, 3, dtype='datetime64[ns]'), arange(0, 3, dtype='datetime64[ns]'))

TypeError: Iterator operand 0 dtype could not be cast from dtype('<M8[ns]') to dtype('<M8[D]') according to the rule 'safe'

In [7]: busday_count(arange(0, 3, dtype='datetime64[D]'), arange(0, 3, dtype='datetime64[D]'))
Out[7]: array([0, 0, 0])
```

**Pull request merged**